src/telemetry.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  defaultStackParser,
+  Hub,
+  makeMain,
+  makeNodeTransport,
+  NodeClient,
+  runWithAsyncContext,
+} from '@sentry/node';
+import packageJson from '../package.json';
+
+export async function withTelemetry<F>(
+  callback: () => F | Promise<F>,
+  enabled: boolean,
+  integration: string,
+): Promise<F> {
+  const { sentryHub, sentryClient } = createSentryInstance(
+    enabled,
+    integration,
+  );
+
+  makeMain(sentryHub);
+
+  const transaction = sentryHub.startTransaction({
+    name: 'sentry-wizard-execution',
+  });
+  sentryHub.getScope().setSpan(transaction);
+  const sentrySession = sentryHub.startSession();
+  sentryHub.captureSession();
+
+  try {
+    return await runWithAsyncContext(() => callback());
+  } catch (e) {
+    sentryHub.captureException('Error during wizard execution.');
+    transaction.setStatus('internal_error');
+    sentrySession.status = 'crashed';
+    throw e;
+  } finally {
+    transaction.finish();
+    sentryHub.endSession();
+    await sentryClient.flush(3000);
+  }
+}
+
+function createSentryInstance(enabled: boolean, integration: string) {
+  const client = new NodeClient({
+    dsn: 'https://8871d3ff64814ed8960c96d1fcc98a27@o1.ingest.sentry.io/4505425820712960',
+    enabled: enabled,
+
+    tracesSampleRate: 1,
+    sampleRate: 1,
+
+    release: packageJson.version,
+    integrations: [],
```

Comment:
Since we're initializing the client here, we need to at least add some integrations back, right? GlobalHandlers, Http?

Reply:
Http: yes. GlobalHandlers I wouldn't add because we try catch in the wrapper anyhow.